src/Integration.Vsix/Analysis/IssueConsumerFactory.cs

```diff
@@ -0,0 +1,71 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Text;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Core.Suppression;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.Analysis
+{
+    /// <summary>
+    /// Callback used by the issue consumer to notify when a new snapshot
+    /// is available i.e. the set of issues has changed.
+    /// </summary>
+    internal delegate void PublishSnapshot(IIssuesSnapshot issuesSnapshot);
```

Comment:
The name implies this class knows what the caller is going to do with this information. It is better IMO to instead announce what happened, i.e. "HandleIssuesChanged", "IssuesChangedHandler", etc, WDYT?

Reply:
Ok. We already have an `HandleNewIssues` method elsewhere; I'll make this `SnapshotChangedHandler` to avoid confusion.